scripts/render_DESeq2_report.parallel.R

```diff
@@ -162,59 +228,26 @@ make_pathway_reports <- function(pars, facet)  {
   }
 }
 
-# Generic function to build and filter facets
-get_facets <- function(metadata = DESeqDesign,
-                       exclude = params$exclude_groups,
-                       display_facet = params$display_group_facet,
-                       skip_extra = "DMSO") {
-  facets <- metadata %>%
-    filter(!(!!sym(display_facet)) %in%
-             c(exclude, skip_extra)) %>%
-    pull(display_facet) %>% 
-    unique()
-  message(paste0("Making multiple reports based on ",
-                 display_facet ,"..."))
-  return(facets)
-}
-
-# Determine whether facets are needed
-# And if so, what they should be
-  # Case 1: no facet, no display facet
-if(is.na(params$group_facet) && is.na(params$display_group_facet)){
-  facets <- NULL
-  # Case 2: no facet, yes display facet
-} else if(is.na(params$group_facet) && !is.na(params$display_group_facet)){
-  facets <- get_facets()
-  # Case 3: yes facet, yes display facet
-} else if(!is.na(params$group_facet) && !is.na(params$display_group_facet)){
-  if(params$group_facet != params$display_group_facet) {
-    stop("Error: display_group_facet must match group_facet, otherwise DESeq2 results get mixed and matched.")
-  }
-  facets <- get_facets()
-  # Which facets have DEGs?
-  hasDEGs <- names(which(sapply(X = mergedDEGsList,
-                                FUN = function(i) length(i)>=1),
-                         arr.ind = T))
-  facets <- facets[facets %in% hasDEGs]
-  # Case 4: yes facet, no display facet, this one doesn't make sense
-} else {
-  stop("Making a single report for faceted data not supported. Did you forget to set display_group_facet?")
-}
-
 if (params$parallel){
   biocluster <- BiocParallel::MulticoreParam(workers = round(params$cpus*0.9))
-  BiocParallel::bpmapply(FUN = make_main_reports, facet = facets, MoreArgs = list(pars = params), BPPARAM = biocluster)
-  if (params$generate_main_report == T) {Sys.sleep(60)}
-  BiocParallel::bpmapply(FUN = make_data_reports, facet = facets, MoreArgs = list(pars = params), BPPARAM = biocluster)
-  if (params$generate_data_explorer_report == T) {Sys.sleep(60)}
-  BiocParallel::bpmapply(FUN = make_pathway_reports, facet = facets, MoreArgs = list(pars = params), BPPARAM = biocluster)
-  if (params$generate_go_pathway_report == T) {Sys.sleep(60)}
-  BiocParallel::bpmapply(FUN = make_stats_reports, facet = facets, MoreArgs = list(pars = params), BPPARAM = biocluster)
+  BiocParallel::bpmapply(FUN = make_main_reports, facet = display_facets, MoreArgs = list(pars = params), BPPARAM = biocluster)
```

Comment:
I think I would suggest reverting these changes (I'm surprised rebase wouldn't have fixed this?) because otherwise the sleep(60) will happen even if the report isn't being generated